### PR TITLE
Added a normalization term to ctc_beam_search_decoder

### PR DESCRIPTION
**Pull request by vsl9**
Added a normalization term to ctc_beam_search_decoder in order to get correct log probabilities.
It solves https://github.com/tensorflow/tensorflow/issues/6034

The decoder is supposed to get logits as inputs instead of log probabilities, so it should compute log(softmax(logits)) inside.
The original code was relied on an equation: log(exp(x)) = x. But it didn't include a softmax normalization term: sum(exp(x_j)).

### CODE REVIEW

File: tensorflow/core/util/ctc/ctc_beam_search.h [hunk 1]
```diff
@@ -259,6 +259,16 @@ void CTCBeamSearchDecoder<CTCBeamState, CTCBeamComparer>::Step(
   } else {
     max_coeff = raw_input.maxCoeff();
   }
+
+  // Get normalization term of softmax: log(sum(exp(logit[j]-max_coeff))).
+  float logsumexp = 0.0;
+  for (int j = 0; j < raw_input.size(); ++j) {
+    logsumexp += expf(raw_input(j) - max_coeff);
```
Comment (ebrevdo) [inline tensorflow/core/util/ctc/ctc_beam_search.h]: can you use std::exp and std::log?  or probably better, Eigen::numext::{log,exp}?

**Comment by ebrevdo** [comment]
This PR looks right; and may have fixed a bug in how we handle uncentered logits.  We're discussing it and will have more to say in a little bit.

**Review by ebrevdo — CHANGES_REQUESTED** [review]

**Comment by vsl9** [comment]
@ebrevdo Sure, I switched to Eigen and updated ground truth values in Keras CTC beam search test.

**Review by ebrevdo — APPROVED** [review]

**Comment by ebrevdo** [comment]
This is perfect, thanks!
